algorithms/linfa-bayes/src/base_nb.rs

```diff
@@ -48,8 +46,17 @@ where
     }
 }
 
+// Trait computing log-likelihood for Naive Bayes models
+pub(crate) trait NaiveBayesLogLikelihood<F, L>
+where
+    F: Float,
+    L: Label + Ord,
+{
+    fn joint_log_likelihood(&self, x: ArrayView2<F>) -> HashMap<&L, Array1<F>>;
+}
+
```

Comment:
No need to have a separate trait for log likelihood.

Reply:
@YuhanLiin Sure, I've moved log-likelihood back into the `NaiveBayes<'a, F, L, D>` trait. The only side effect is that now one needs to explicitly specify type parameter D when calling `joint_log_likelihood`, e.g. here:  https://github.com/rust-ml/linfa/pull/190/files#diff-20110e51d7148056af774f667a88af96f6abf6f9f3d3e79961e2098b6f11f441R261-R262. Compiler can't automatically infer type `D`  because `joint_log_likelihood` doesn't depend on it, but the whole `NaiveBayes` trait does.

Reply:
In `NaiveBayes` can you move type parameter `D` from the trait to the `predict_inplace` method, since it's only relevant for that method? That should fix the type inference issues.

Reply:
Yep, that worked, thanks